### ISSUE TITLE
MultiOutputStream

### ISSUE THREAD

**Issue by mikegolod** [google/guava#2485]
We have MultiInputStream, why don't we have a MultiOutputStream?

**Comment by cpovirk** [comment]
We have one internally, but it has never been very popular. I assume you're looking for a stream that forwards writes to several other streams?

**Comment by mikegolod** [comment]
Exactly! I've been in the process of cleaning dependencies for one project i'm working on. Apache Commons IO have TeeOutputStream class. It's the only class from that whole library my project used. I prefer to use guava instead of apache libs. So i wrote MultiOutputStream myself for now.

**Comment by vharuty** [comment]
@cpovirk hi, please let me I know what is the status for this issue?
I saw the issue and have an idea how to implement it.
Please let me know.

**Comment by cpovirk** [comment]
We do have one implemented. We just need to decide if it's worth releasing (and if it needs any API changes, like switching to using suppressed exceptions when available). If you need one for yourself, though, you probably should stick with your own. We might take a long time :(

**Comment by vharuty** [comment]
@cpovirk Just out of curiosity, does your implementation of "write" method of the mentioned class throw an "aggregate" exception containing the exception(s) occurred on OutputStream(s).
If yes then does the aggregated exception contain reference to faulting OutputStream?

We have implementation like.
```
  public class OutputStreamException extends IOException {
    private static final long serialVersionUID = 4271756986146326876L;
    private OutputStream stream;
    public OutputStream getFaultingStream(){
      return stream;
    }
    public OutputStreamException(Throwable ex, OutputStream stream){
      super(ex);
      this.stream = stream;
    }
  }
  
  public class MultiOutputStreamAggregateException extends IOException {
    private static final long serialVersionUID = -3124926231957260300L;
    private Iterable<? extends IOException> innerExceptions;
    public Iterable<? extends IOException> getInnerExcpetions() {
      return innerExceptions;
    }
    public MultiOutputStreamAggregateException(Iterable<? extends IOException> innerExceptions){
      super("Aggregated MultiOutputStream IO exception");
      this.innerExceptions = innerExceptions;
    }
  }
```

Thanks

**Comment by cpovirk** [comment]
We do throw an aggregate `IOException`. We don't provide a reference back to the particular `OutputStream` that failed, though. I hadn't thought about that.
